### PR TITLE
[NT-1704] Update ReactiveSwift to 6.5.0

### PR DESCRIPTION
**Pull request by justinswart**
# 📲 What

Updates ReactiveSwift to `6.5.0`.

**Note:** https://github.com/kickstarter/Kickstarter-ReactiveExtensions/pull/82 should merge before this.

# 🤔 Why

There have been a number of improvements to this framework that we use throughout our codebase (see release notes below).

# 🛠 How

- Updated our `Cartfile`.
- Replaced deprecated instances of `filterMap` with `compactMap`.
- Updated CircleCI to use Xcode 12.2 and iOS 14.2.
- Fixed Carthage workaround script.

# 👀 See

[ReactiveSwift release notes](https://github.com/ReactiveCocoa/ReactiveSwift/releases)
[Carthage workaround](https://github.com/Carthage/Carthage/blob/master/Documentation/Xcode12Workaround.md)

# ✅ Acceptance criteria

- [ ] Build and tests pass

### CODE REVIEW

File: Cartfile [hunk 1]
```diff
@@ -1,15 +1,15 @@
 ### Internal
 
 github "kickstarter/Kickstarter-Prelude" "9118beee2b37fa5fc5a4f56502c1fa99030b9f9e"
-github "kickstarter/Kickstarter-ReactiveExtensions" "48a3f507995a6bdae964ddf9f2347af6bf74a19f"
+github "kickstarter/Kickstarter-ReactiveExtensions" "807153fe7e2cdb9ebcdc46862458e9e2c6b70d94"
```
Comment (justinswart) [inline Cartfile:4]: This SHA will be updated once https://github.com/kickstarter/Kickstarter-ReactiveExtensions/pull/82 merges.

File: bin/carthage.sh [hunk 2]
```diff
@@ -10,26 +10,17 @@ set -euo pipefail
 xcconfig=$(mktemp /tmp/static.xcconfig.XXXXXX)
 trap 'rm -f "$xcconfig"' INT TERM HUP EXIT
 
+
 # For Xcode 12 make sure EXCLUDED_ARCHS is set to arm architectures otherwise
 # the build will fail on lipo due to duplicate architectures.
-# Xcode 12 Beta 3:
```
Comment (justinswart) [inline bin/carthage.sh:15]: These were no longer needed for the workaround and caused Alamofire to fail to build.

**Comment by justinswart** [comment]
@singhhari blocking this until #1352 merges - it appears that iOS 14.2 may require us to re-record all of our snapshots 🤦‍♂️

File: Library/ViewModels/DiscoveryPageViewModel.swift [hunk 3]
```diff
@@ -398,9 +398,7 @@ public final class DiscoveryPageViewModel: DiscoveryPageViewModelType, Discovery
       .skipNil()
 
     self.dismissPersonalizationCell = self.personalizationCellDismissTappedProperty.signal
-      .on { _ in
-        AppEnvironment.current.userDefaults.hasDismissedPersonalizationCard = true
-      }
+      .on(value: { _ in AppEnvironment.current.userDefaults.hasDismissedPersonalizationCard = true })
```
Comment (singhhari) [inline Library/ViewModels/DiscoveryPageViewModel.swift:401]: Interesting. The update to 6.5 wants the value param explicitly written now?

**Review by singhhari — APPROVED** [review]
lgtm

File: Library/ViewModels/DiscoveryPageViewModel.swift (hunk 3 above)
Comment (justinswart) [reply Library/ViewModels/DiscoveryPageViewModel.swift:401]: It works without it but complains about ambiguity.

**Pull request merged**